types.go

```diff
@@ -257,3 +257,24 @@ func resize(x []float64, dim int) []float64 {
 	}
 	return x[:dim]
 }
+
+func resizeDense(m *mat64.Dense, r, c int) *mat64.Dense {
```

Comment:
Use Dense.View for this.

The other types don't have that though, so leave them until we support that.

Note @btracey, this use of View is not supported as well as it used to be because View returns a Matrix(*Dense) and this could all be done in place.